### PR TITLE
Integrate gutenberg-mobile release 1.55.1

### PR DESCRIPTION
**Pull request by fluiddot**
## Description
This PR incorporates the 1.55.1 release of gutenberg-mobile.
For more information about this release and testing instructions, please see the related Gutenberg-Mobile PR: https://github.com/wordpress-mobile/gutenberg-mobile/pull/3632

Release Submission Checklist

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: Podfile [hunk 1]
```diff
@@ -161,7 +161,7 @@ abstract_target 'Apps' do
     ## Gutenberg (React Native)
     ## =====================
     ##
-    gutenberg :tag => 'v1.55.0'
+    gutenberg :commit => '396d23c243956729cf412fbd47366f7e8e63aad9'
```
Comment (mokagio) [inline Podfile]: @fluiddot Do you plan to wait to merge this till `v1.55.1` lands and its tag has been pushed to the GitHub?

Comment (fluiddot) [reply Podfile]: Yeah, this PR will only be merged after the `v1.55.1` tag is created.

**Review by antonis — APPROVED** [review]
Tested via build 50766 from AppCenter and works as expected. The changes LGTM 🎉

**Comment by mokagio** [comment]
This PR has been shipped as part of [17.6 beta 3](https://github.com/wordpress-mobile/WordPress-iOS/pull/16718).

**Pull request merged**